### PR TITLE
refactor(users): ensure loading indicators on POST

### PR DESCRIPTION
**Pull request by jniles**
This commit hooks up the `bhSubmit` and `bhLoadingButtons` to properly
show a loading indicator when data is POSTed on the users and permission
page.

-----

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: client/src/js/services/UserService.js [hunk 1]
```diff
@@ -79,4 +80,8 @@ function UserService($http, util) {
     return $http.put('/users/' + id + '/password', data)
     .then(util.unwrapHttpResponse);
   }
+
+  function validatePassword(passwordA, passwordB) {
+    return passwordA && passwordA && passwordA === passwordB;
```
Comment (sfount) [inline client/src/js/services/UserService.js]: This little method could really use a comment for future maintainers!

Comment (sfount) [reply client/src/js/services/UserService.js]: Is it supposed to include a length check? It seems like that code has been moved but I can't verify the functionality is equivalent.

**Review by sfount — COMMENTED** [review]
A good user experience improvement. I've made a comment on one of the utility methods here that it would be good to get a comment on. Otherwise this looks good to merge.

File: client/src/js/services/UserService.js (hunk 1 above)
Comment (jniles) [reply client/src/js/services/UserService.js]: Whoops!  Good catch!

**Comment by jniles** [comment]
@sfount, changes have landed.

**Comment by sfount** [comment]
Changes look good. This can be merged when the tests pass.

File: client/src/js/services/UserService.js [hunk 2]
```diff
@@ -79,4 +80,19 @@ function UserService($http, util) {
...
     .then(util.unwrapHttpResponse);
   }
+
+  /**
+   * @function validatePassword
+   *
+   * @param {String} passwordA - a user password
+   * @param {String} passwordB - a challenge password
+   *
+   * @description
```
Comment (sfount) [inline client/src/js/services/UserService.js:90]: :+1:

**Pull request merged**